lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -422,19 +422,9 @@ def tab_complete_stub(str, quote: nil)
       end
     }
 
-    # Verify that our search string is a valid regex
-    begin
-      Regexp.compile(str,Regexp::IGNORECASE)
-    rescue RegexpError
-      str = Regexp.escape(str)
-    end
-
-    # @todo - This still doesn't fix some Regexp warnings:
-    # ./lib/rex/ui/text/dispatcher_shell.rb:171: warning: regexp has `]' without escape
-
     # Match based on the partial word
     items.find_all { |word|
-      word.downcase.start_with?(str.downcase) || word =~ /^#{str}/i
+      word.downcase.start_with?(str.downcase)
```

Comment:
One last bread crumb trail for why this was deleted; It was thought that the old regex approach for finding strings that start with search term was a left over perl-ism, and `word.downcase.start_with?(str.downcase)` is now being used as the preferred Ruby pattern. If there's community raised issues that this breaks workflows, we can re-evaluate it in the future :+1: